Supporto a Django 2.x + autocomplete

udanieli opened the pull request:
Timido tentativo di rendere compatibile l'app per django 2.0 (vedi #2) + supporto alla nuova libreria `django-autocomplete-light`.

* model: aggiunte regole `on_delete` obbligatorie per django 2.0
* autocomplete: rimossi import e funzioni ormai obsolete
* autocomplete: aggiunte Class-Based Views e URLconf come da [tutorial](https://django-autocomplete-light.readthedocs.io/en/master/tutorial.html)
* aggiornato README.md con esempio per autocompletamento

Consiglio un branch separato se si vuole mantenere la compatibilità con le vecchie versioni se no si scassa tutto :(

udanieli committed:
rendi compatibile per django 2.0

udanieli committed:
update modulo autocomplete obsoleto

la libreria django-autocomplete-light è cambiata pesantemente
TODO: update wiki

udanieli committed:
update README.md

kbytesys commented:
Ciao!

Guardo la PR nel weekend. Non è tuttavia un problema rinunciare alla compatibilità con i vecchi django, si può pensare al limite di aumentare la versione del package in modo da rendere chiaro il salto di versione.

udanieli commented on comuni_italiani/autocomplete/apps.py:
```diff
@@ -6,59 +6,3 @@
-                                    search_fields=['^name'],
-                                    attrs={
-                                        'placeholder': 'Nome della città',
-                                        'data-autocomplete-minimum-characters': 2,
```
i caratteri minimi per la ricerca ora vanno negli [attributi del widget](https://django-autocomplete-light.readthedocs.io/en/master/tutorial.html#passing-options-to-select2) quando decidiamo di usarlo in un form.

udanieli commented on comuni_italiani/autocomplete/apps.py:
```diff
@@ -6,59 +6,3 @@
-                                    },
-                                    widget_attrs={
-                                        'data-widget-maximum-values': getattr(
-                                            settings, "COMUNI_ITALIANI_AUTOCOMPLETE_RESULTS", 10),
```
Ho tirato via il limite dato che [django-autocomplete-light]() fa già la paginazione, il limite si imposta come attributo `paginate_by` (default = 10) nella CBV.

udanieli committed:
numero custom risultati paginazione autocomplete

udanieli committed:
usa nuovo metodo path() in URLconf

La doc django dice che url() diverrà deprecated:

This function is an alias to django.urls.re_path().
It’s likely to be deprecated in a future release
